src/main/java/com/minecolonies/colony/Field.java

```diff
@@ -0,0 +1,467 @@
+     * @param world the world object.
+     * @param position the position.
+     * @return true if it is.
+     */
+    public boolean isNoPartOfField(World world, BlockPos position)
+    {
+        return world.isAirBlock(position) || world.getBlockState(position.up()).getBlock().getMaterial().isSolid();
+    }
+
+    /**
+     * Returns the {@link BlockPos} of the current object, also used as ID
+     *
+     * @return          {@link BlockPos} of the current object
+     */
+    public BlockPos getID()
+    {
+        // Location doubles as ID
+        return this.location;
+    }
+
+    /**
+     * Returns the colony of the field
+     *
+     * @return          {@link com.minecolonies.colony.Colony} of the current object
+     */
+    public Colony getColony()
+    {
+        return this.colony;
+    }
+
+    /**
+     * Create and load a Field given it's saved NBTTagCompound
+     *
+     * @param colony    The owning colony
+     * @param compound  The saved data
+     * @return          {@link com.minecolonies.colony.Field} created from the compound.
+     */
+    public static Field createFromNBT(Colony colony, NBTTagCompound compound)
+    {
+        Field field = new Field(colony);
+        field.readFromNBT(compound);
+        return field;
+    }
+
+    /**
+     * Save data to NBT compound.
+     * Writes the {@link #location} value.
+     *
+     * @param compound      {@link net.minecraft.nbt.NBTTagCompound} to write data to
+     */
+    public void writeToNBT(NBTTagCompound compound)
+    {
+        BlockPosUtil.writeToNBT(compound, TAG_LOCATION, this.location);
+        compound.setBoolean(TAG_TAKEN, taken);
+        compound.setInteger(TAG_STAGE, fieldStage.ordinal());
+        compound.setInteger(TAG_LENGTH_PLUS, lengthPlusX);
+        compound.setInteger(TAG_WIDTH_PLUS, widthPlusZ);
+        compound.setInteger(TAG_LENGTH_MINUS, lengthMinusX);
+        compound.setInteger(TAG_WIDTH_MINUS, widthMinusZ);
+        inventory.writeToNBT(compound);
+    }
+
+    /**
+     * Save data to NBT compound.
+     * Writes the {@link #location} value.
+     *
+     * @param compound      {@link net.minecraft.nbt.NBTTagCompound} to write data to
+     */
+    public void readFromNBT(NBTTagCompound compound)
+    {
+        location = BlockPosUtil.readFromNBT(compound, TAG_LOCATION);
+        taken = compound.getBoolean(TAG_TAKEN);
+        fieldStage = FieldStage.values()[compound.getInteger(TAG_STAGE)];
+        lengthPlusX = compound.getInteger(TAG_LENGTH_PLUS);
+        widthPlusZ = compound.getInteger(TAG_WIDTH_PLUS);
+        lengthMinusX = compound.getInteger(TAG_LENGTH_MINUS);
+        widthMinusZ = compound.getInteger(TAG_WIDTH_MINUS);
+        inventory = new InventoryField("Scarecrow", true);
+        inventory.readFromNBT(compound);
+    }
+
+    /**
+     * Has the field been taken?
+     * @return true if the field is not free to use, false after releasing it.
+     */
+    public boolean isTaken() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this left curly brace to the beginning of next line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)